source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -372,10 +370,15 @@ namespace PlayFab
     void PlayFabWinHttpPlugin::SetPredefinedHeaders(const CallRequestContainer& requestContainer, HINTERNET hRequest)
     {
         UNREFERENCED_PARAMETER(requestContainer);
-        WinHttpAddRequestHeaders(hRequest, L"Accept: application/json", -1, 0);
-        WinHttpAddRequestHeaders(hRequest, L"Content-Type: application/json; charset=utf-8", -1, 0);
-        WinHttpAddRequestHeaders(hRequest, (L"X-PlayFabSDK: " + std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end())).c_str(), -1, 0);
-        WinHttpAddRequestHeaders(hRequest, L"X-ReportErrorAsSuccess: true", -1, 0);
+        setPredefinedHeadersFailed = TryAddHeader(hRequest, L"Accept: application/json") &&
+            TryAddHeader(hRequest, L"Content-Type: application/json; charset=utf-8") &&
+            TryAddHeader(hRequest, (L"X-PlayFabSDK: " + std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end())).c_str()) &&
+            TryAddHeader(hRequest, L"X-ReportErrorAsSuccess: true");
+    }
+
+    BOOL PlayFabWinHttpPlugin::TryAddHeader(HINTERNET hRequest, LPCWSTR lpszHeaders)
```

Comment:
>BOOL [](start = 4, length = 4)

nit: I prefered this as a normal `bool`. putting a dependency on win data types in your public headers (even for a windows specific plugin) is unnecessary